pkg/k8s/client.go

```diff
@@ -0,0 +1,76 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type client struct {
+	clientset kubernetes.Interface
+}
+
+func (c client) do(ctx context.Context, secret *apiv1.Secret) error {
+	_, err := c.createSecret(ctx, secret)
```

Comment:
should we try getting the secret first and just creating it if it returns an error?

Reply:
I am not sure of the answer to this one.  The intent of this work was to keep the existing behavior in place while improving the error handling / flow of the code itself.  I wonder why the flow was originally setup this way, and if there was some documentation / recommendation on creating the secret as the default first action.